src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,146 @@ class GetinfoRequestHandler: public BaseRequestHandler
     }
 };
 
+/** Process netinfo requests */
+class NetinfoRequestHandler : public BaseRequestHandler
+{
+private:
+    enum struct m_conn_type {
+        ipv4,
+        ipv6,
+        onion,
+    };
+
+    struct m_peer {
+        int id;
+        int mapped_as;
+        double min_ping;
+        double ping;
+        std::string addr;
+        m_conn_type conn_type;
+        bool is_outbound;
+        bool is_block_relay;
+        bool operator<(const m_peer& rhs) const { return std::tie(is_outbound, min_ping) < std::tie(rhs.is_outbound, rhs.min_ping); }
+    };
```

Comment:
I think adding user agent in verbose mode is worthy. Beyond mode of display, how do you see the data split between this new RPC and `getpeerinfo` ? Static-set-at-once-at-connection-opening versus dynamic-messages-processed ?

Reply:
By user agent, do you mean getpeerinfo.subver? aka `cleanSubVer` in net.h

Reply:
added version+subver together in a column